gstvstaudioprocessor.cpp

```diff
@@ -1137,16 +1148,121 @@ create_element_name(const gchar * prefix, const gchar * class_name)
   return element_name;
 }
 
+#if defined(__linux__) && !defined(__BIONIC__)
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  gst_plugin_add_dependency_simple (plugin, NULL, "/usr/lib/vst3:/usr/local/lib/vst3", ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+  gst_plugin_add_dependency_simple (plugin, "HOME/.vst3", NULL, ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+}
+#elif defined(G_OS_WIN32)
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  PWSTR wideStr {};
+  char commonpath[MAX_PATH];
+  gchar *path;
+  if (FAILED (SHGetKnownFolderPath (FOLDERID_ProgramFilesCommon, 0, NULL, &wideStr)))
+    return;
+
+  if (!WideCharToMultiByte(CP_ACP, 0, wideStr, -1, commonpath, MAX_PATH, NULL, NULL))
+    return;
+
+  path = g_build_filename (commonpath, "VST3", NULL);
+  gst_plugin_add_dependency_simple (plugin, NULL, path, ".vst3",
+      (GstPluginDependencyFlags) (GST_PLUGIN_DEPENDENCY_FLAG_RECURSE |
+      GST_PLUGIN_DEPENDENCY_FLAG_FILE_NAME_IS_SUFFIX));
+  g_free (path);
+}
+#else
+static void
+register_system_dependencies(GstPlugin *plugin)
+{
+  GST_FIXME_OBJECT (plugin, "Implement plugin dependencies support for this platform");
+}
+#endif
+
+static void
+list_paths_with_vst3_extension (VST3::Hosting::Module::PathList &paths,
+    const gchar *path, gboolean recurse)
+{
+  GDir *dir;
+  const gchar *dirent;
+  gchar *filename;
+
+  dir = g_dir_open (path, 0, NULL);
+  if (!dir)
+    return;
+
+  while ((dirent = g_dir_read_name (dir))) {
+    GStatBuf file_status;
+
+    filename = g_build_filename (path, dirent, NULL);
+    if (g_stat (filename, &file_status) < 0) {
+      g_free (filename);
+      continue;
+    }
+
+    if (g_str_has_suffix (dirent, ".vst3"))
+      paths.push_back (filename);
+
+    if ((file_status.st_mode & S_IFDIR) && recurse)
+      list_paths_with_vst3_extension (paths, filename, recurse);
+
+    g_free (filename);
+  }
+
+  g_dir_close (dir);
+}
+
 void
 gst_vst_audio_processor_register(GstPlugin * plugin)
 {
+  const gchar *main_exe_path;
+  const gchar *paths_env_var;
+
   GST_DEBUG_CATEGORY_INIT (gst_vst_audio_processor_debug, "vst-audio-processor", 0,
       "VST Audio Processor");
 
+  register_system_dependencies(plugin);
+
   audio_processor_info_quark = g_quark_from_static_string ("gst-vst-audio-processor-info");
 
   auto paths = VST3::Hosting::Module::getModulePaths();
 
+#ifdef HAVE_GST_EXE_PATH
+  main_exe_path = gst_get_main_executable_path ();
+  if (main_exe_path) {
+    gchar *vst3_exe_path = g_build_filename (main_exe_path, "vst3", NULL);
```

Comment:
Lowercase `vst` or uppercase?

Reply:
lowercase on Linux, uppercase on Windows and Mac, gaah